src/components/execution/AddExecutionForm.js

```diff
@@ -175,10 +194,31 @@ class AddExecutionForm extends Component {
     this.setState({ userProvidedFilename: '' });
   };
 
+  executePipeline = () => {
+    const { pipelines, dispatchExecutePipeline } = this.props;
+    const {
+      sourceId,
+      userProvidedFilename,
+      parameters,
+      schemaId,
+      pipelineId,
+    } = this.state;
+
+    const pipeline = pipelines.find((pipe) => pipe.id === pipelineId);
+    dispatchExecutePipeline({
+      pipeline,
```

Comment:
Since we already have the entire pipelines fetched on the frontend, providing the pipelineId to the listener to then re-fetch all the pipelines and then finding the right one does not provide any extra value to the listener structure since it works seamlessly with the frontend.